### ISSUE TITLE
Storage Access Framework

### ISSUE THREAD

**Issue by babaric-dev** [termux/termux-api#544]
**Feature description**
Nowadays, many people use external sdcards and I think it will be convenient to have a feature to have write access through termux-api.

**Reference implementation**
[Storage Access Framework](https://developer.android.com/guide/topics/providers/document-provider)

**Comment by tareksander** [comment]
Do you need any more features than already implemented in #475 ?

**Comment by babaric-dev** [comment]
Wait.. Is it already implemented? I got a command not found for `termux-saf-read`.

**Comment by tareksander** [comment]
It is implemented, but there wasn't a release since it was implemented. I don't know if termux-saf-* is included in the package yet since they won't work, but you can view the help text in the source code (or just download the repo and run the scripts): https://github.com/termux/termux-api-package/tree/master/scripts

**Comment by babaric-dev** [comment]
Thank you. You mentioned saf will not work and told me to have a try of the scripts. So, it simply wont work?

**Comment by tareksander** [comment]
It won't work unless you use the github builds, where the functionality is already implemented

**Comment by babaric-dev** [comment]
If I use the Termux:API from GH build, do I need to have to use the Termux build fron GH?

**Comment by tareksander** [comment]
Yes, Termux and all plugins must be installed from the same source.

**Comment by babaric-dev** [comment]
Great. I hope an update for termux api and the apt package will be released ASAP. (Termux with SAF, cool, isnt it?)
